sdk/core/core-amqp/src/retry.ts

```diff
@@ -173,49 +173,56 @@ function calculateDelay(
  */
 export async function retry<T>(config: RetryConfig<T>): Promise<T> {
   validateRetryConfig(config);
-  if (!config.retryOptions) {
-    config.retryOptions = {};
+  const updatedConfig = config;
```

Comment:
This may just be a reference to `config`? I ran a little test and find that changing property of `b` also updates `a`, while using `{ ...a }` the result is expected

```
> const a = { o: 1 }
undefined
> const b = a;
undefined
> const c = { ...a }
undefined
> c.o = 3
3
> a
{ o: 1 }
> c
{ o: 3 }
> b.o = 2
2
> a
{ o: 2 }
> b
{ o: 2 }
```

Reply:
Fixed it!